### PR TITLE
Prefix BaseItem class members

### LINKED ISSUE #233: Prefix a _ to all methods in item classes
**I certify I'm including:**

- [x] A descriptive title
- [x] A description of the problem I'm trying to solve, including *why* I think this is a problem
- [x] An overview of the suggested solution
- [ ] If the feature changes current behaviour, reasons why my solution is better

Currently if an aggregate defines a field called `execute` (or any of the other methods we have in the item classes) the evaluation code will return the` execute method` as opposed to the `execute field`. To prevent this kind of a problem all methods and variables should start with _ (even if they are not private) so as the completely separate out the namespace of internal names.

We already don't support `Field` names starting with _ for this very problem.

### PR DESCRIPTION
**Pull request by prashastk**
#### Pull Request Checklist:
- [x] Tests for the changes have been added
- [x] Docs have been added / updated

### What kind of change does this PR introduce?
Prefix BaseItem class members with `_` for internal use and `run_` for external use. 
Also, stopped inhering from BaseItem for the Anchor class.

#### Link to issue
fixes #233

### CODE REVIEW

File: blurr/runner/local_runner.py [hunk 1]
```diff
@@ -44,7 +44,8 @@ def get_identity_records_from_json_files(
         identity_records = defaultdict(list)
         for file in local_json_files:
```
Comment (indradeep) [inline blurr/runner/local_runner.py]: Passing thought, in cases like these, I use the initial `f` as part of the for loop as an index, and use the term `file` as the object i'm dealing with in the rest of the code.  Makes the function call parameters more readable :)

File: tests/core/base_item_test.py [hunk 2]
```diff
@@ -67,3 +69,37 @@ def test_base_item_filter_missing(schema_spec: Dict[str, Any]) -> None:
     test_item = get_test_item(schema_spec)
 
     assert test_item._needs_evaluation
+
+
+def test_base_item_method_naming():
```
Comment (indradeep) [inline tests/core/base_item_test.py]: Ultra cool!

File: blurr/runner/local_runner.py (hunk 1 above)
Comment (prashastk) [reply blurr/runner/local_runner.py]: renamed f to file_stream. That should make it much more clear.

File: tests/core/base_item_test.py (hunk 2 above)
Comment (prashastk) [reply tests/core/base_item_test.py]: I know! I was quite pleased with this myself :)

**Pull request merged**